NuRadioReco/modules/voltageToAnalyticEfieldConverterNeutrinos.py

```diff
@@ -0,0 +1,546 @@
+					phased_trace += trace_phasing
+			return phased_trace
+
+		prop = propagation.get_propagation_module(name = 'analytic')
+		antenna_position = det.get_relative_position(station_id, channel_Vpol)
+		r = prop(vertex_position, antenna_position, icemodel, 
+				 attenuation_model = attenuation_model)
+		r.find_solutions()
+		n = r.get_number_of_solutions()
+		for iS in range(n):
+			
+			if r.get_solution_type(iS) == ray_solution:
+				R = r.get_path_length(iS)
+				attn = r.get_attenuation(iS, ff, 0.5*sampling_rate)
+
+				
+		time_trace = get_time_trace(10**18,viewing_angle, N_V_time_domain, 
+									dt, shower_type, n_index, R, parametrization) 
+		trace_freq = fft.time2freq(time_trace, 1/dt)
+
+		if attenuation:
+				trace_freq *= attn
+		if bandpass:
+				trace_freq *= h
+		
+		analytic_trace_fft = np.sum(efield_antenna_factor[use_channels_efield.index(channel_Vpol)]*np.array([trace_freq,
+																											 trace_freq]), axis = 0)
+		for iCh, trace in enumerate(V_timedomain):
+			if use_channels_efield[iCh] == channel_Vpol: # use Vpol to determine offset
+				if phasing: trace = get_phased_trace(use_channels_efield[iCh], phasing_dict[use_channels_efield[iCh]])
+				if bandpass:
+					f = fft.time2freq(trace, 1/dt) * h
+					trace = fft.freq2time(f, 1/dt)
+				
+				analytic_trace = fft.freq2time(analytic_trace_fft, 1/dt)
+	
+				corr = signal.hilbert(signal.correlate(trace, analytic_trace))
+
+				toffset = np.argmax(corr) - (len(corr)/2) 
+		
+		options = {'maxiter':500, 'disp':True}
+		bnds = ((10**17, 10**20))
+		results = opt.minimize(minimizer_theta_component, x0 = (10**18), 
+							   method = 'Nelder-Mead', args = (R, n_index, toffset),
+							   bounds = bnds, options = options) #Fit amplitude theta component with viewing angle fixed to 55 degrees
+
+		bnds = ((10**17, 10**19), (np.deg2rad(40), np.deg2rad(70)))
+		results = opt.minimize(minimizer_theta_component, 
+							   x0 = [results.x[0], np.deg2rad(55)], 
+							   method = 'Nelder-Mead', args = (R, n_index, toffset),
+							   bounds = bnds) #Fit amplitude theta component and viewing angle
+
+		
+		
+		#adapt the time for the fit for the shift of the Hpol wrt Vpol
+		positions = np.array([det.get_relative_position(station_id,channel_Vpol),
+							  det.get_relative_position(station_id,channel_Hpol)])
+		time_delay1 = get_time_delay_from_direction(zenith, azimuth, positions[0], 
+													n = n_index)
+		time_delay2 = get_time_delay_from_direction(zenith, azimuth, positions[1], 
+													n = n_index)
+		time_shift_Hpol = int(-(time_delay2-time_delay1))
+		time_shift_Hpol += toffset
+		
+		bnds = ((10**17, 10**19))
+		results1 = opt.minimize(minimizer, x0=(results.x[0]), method = 'Nelder-Mead', 
+								args = (R, n_index, time_shift_Hpol, False, True), 
+								bounds = bnds, options = options) #Fit amplitude phi component
+
+
+		
+		#store electricfield 
+		time_trace_theta = get_time_trace(results.x[0], results.x[1], N_V_time_domain, 
```

Comment:
Fitting both components to showers with different energies is probably not a good idea. I think a scaling factor would be better